src/lightning-address.ts

```diff
@@ -101,7 +109,7 @@ export default class LightningAddress {
     let callbackUrl = new URL(callback)
     callbackUrl.search = new URLSearchParams(invoiceParams).toString()
 
-    return this.generateInvoice(callbackUrl);
+    return this.generateInvoice(callbackUrl, invoiceParams);
```

Comment:
actually I think we should not pass on the URL here
the object/this knows about the callback URL so we only need to pass the invoiceParams and `generateInvoice` can use the callback URL from `this`